ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -0,0 +1,434 @@
+package controllers
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"sync"
+	"time"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/certs"
+	"github.com/openshift/hypershift/support/releaseinfo"
+	"github.com/openshift/hypershift/support/releaseinfo/registryclient"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// LocalIgnitionProvider is an IgnitionProvider that executes MCO binaries
+// directly to build ignition payload contents out of a given release image and
+// a config string containing 0..N MachineConfig YAML definitions.
+//
+// To do this, MCO binaries and other static input files are extracted from a
+// release image into CacheDir. These contents are cleaned up after each
```

Comment:
why this called CacheDir? what's it caching? shouldn't it just be workdir?

Reply:
Agree, fixed